### DISCUSSION TITLE
Flyway upgrade recommended: MariaDB 10.11 is newer than this version of Flyway

### DISCUSSION THREAD

**Discussion by grappa24** [steve-community/steve#1553] [Q&A] [answered]
sorry, I'm completely new to steve and I ran my first "mvn package" today with the following result.

How can I fix this problem, how can I upgrade my flyway plugin?

Regards, grappa24

`pi@raspberrypi:/steve-steve-3.4.9 $ mvn package
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[INFO] Scanning for projects...
[INFO]
[INFO] -------------------------< de.rwth.idsg:steve >-------------------------
[INFO] Building steve 3.4.9
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-enforcer-plugin:3.0.0:enforce (enforce-java) @ steve ---
[INFO]
[INFO] --- git-commit-id-plugin:4.9.10:revision (default) @ steve ---
[INFO]
[INFO] --- properties-maven-plugin:1.1.0:read-project-properties (default) @ steve ---
[INFO]
[INFO] --- flyway-maven-plugin:7.15.0:migrate (default) @ steve ---
[INFO] Flyway Community Edition 7.15.0 by Redgate
[INFO] Database: jdbc:mysql://192.168.178.135:3306/stevedb (MySQL 5.5)
[WARNING] Flyway upgrade recommended: MariaDB 10.11 is newer than this version of Flyway and support has not been tested. The latest supported version of MariaDB is 10.5.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  35.339 s
[INFO] Finished at: 2024-08-20T16:01:12+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.flywaydb:flyway-maven-plugin:7.15.0:migrate (default) on project steve: org.flywaydb.core.api.exception.FlywayValidateException: Validate failed: Migrations have failed validation
[ERROR] Detected failed migration to version 0.7.7 (update). Please remove any half-completed changes then run repair to fix the schema history.
[ERROR] Need more flexibility with validation rules? Learn more: https://rd.gt/3AbJUZE
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
`

**Comment by goekay** [comment, marked as answer]
hey, you are using a very recent mariadb version that steve does not support.

pls read and follow the readme steps. you will see the following sentence:

> MySQL or MariaDB. You should use [one of these](https://github.com/steve-community/steve/blob/master/.github/workflows/main.yml#L11) supported versions.

**Reply by grappa24** [reply]
I'm using a Synology NAS and I only have the choice between MariaDB 10 or MariaDB 5.
Will Steve work with MariaDB 5 ?
